tests/.infra/crw-ci/dashboard-next/pr-check/Jenkinsfile

```diff
@@ -154,12 +154,17 @@ pipeline {
 
                 stage("Build Che including upstream projects") {
                     steps {
-                        echo "Build branch ${ghprbSourceBranch} of upstream projects"
+                        echo "Install dependencies needed for build of Eclipse Che"
                         sh """
-                            sudo sed -i 's/https/http/g' /etc/yum.repos.d/epel.repo
+                            sudo yum install -y glibc.i686 libfontconfig.so.1 libstdc++-devel.i686 || true
+
+                            # workaround https://github.com/eclipse/che/issues/19093
+                            sudo yum install -y glibc.i686 libfontconfig.so.1 libstdc++-devel.i686 || true
+                            sudo yum install -y glibc.i686 libfontconfig.so.1 libstdc++-devel.i686 || true
```

Comment:
The workaround for "can't install a thing" is "try it 4 times, that'll work!" ??? I mean, cool if it does but how do you know you won't have to add a 5th attempt in future? :D 

What about installing from a different repo, like the pulp ones? or using rhel8? 

Also aside, am I crazy or does i686 == 32 bit? should we be using x86_64 instead?

Reply:
some settings you could try in /etc/dnf/dnf.conf (or yum.conf if you're on rhel7)

```
fastestmirror=True [default false]
max_parallel_downloads=[some value larger than default of 3?]
minrate=[lower value to avoid connection dropping]
retries=[value larger than default of 10]
timeout=[value greater than default of 30 (seconds)]
```

See also `man dnf.conf` or `man yum.conf`

Reply:
> The workaround for "can't install a thing" is "try it 4 times, that'll work!" ??? I mean, cool if it does but how do you know you won't have to add a 5th attempt in future? :D

It does usually take 2-3 iterations to install all three packages, e.g. https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/blue/organizations/jenkins/dashboard-next-happy-path/detail/dashboard-next-happy-path/377/pipeline/41
4th one is reserved, and nothing prevents us to add 5th one if it will be needed.

>  What about installing from a different repo, like the pulp ones? or using rhel8?

I am not so familiar with /etc/yum.repos.d/epel.repo to configure it properly. Anyway, as you can see [in the logs](/etc/yum.repos.d/epel.repo) installation is ending up in pulp repo.

> Also aside, am I crazy or does i686 == 32 bit? should we be using x86_64 instead?

Reply:
1. From your log:

```
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
```

You might be able to enable delta rpms and get better install performance by fetching smaller files :)

2. Assuming this test is building only the dashboard (not all of Che Server), perhaps you could use the docker build instead of installing all the deps?

https://gitlab.cee.redhat.com/codeready-workspaces/crw-jenkins/-/blob/master/jobs/CRW_CI/crw-server_2.7.jenkinsfile#L160

Or if it's also building Che Server, then ...

https://gitlab.cee.redhat.com/codeready-workspaces/crw-jenkins/-/blob/master/jobs/CRW_CI/crw-server_2.7.jenkinsfile#L182-184

Are these glibc and fontconfig deps because you're doing e2e tests that need them? As you can see from the Che build script above, I've had to *disable all the tests* ...

```
mvn -DskipTests clean install ${MVN_FLAGS} -P native -f ${CHE_path}/pom.xml ${MVN_EXTRA_FLAGS}
```
as they wouldn't run (for 2.6, anyway). I haven't re-enabled them yet for 2.7 to see if they're more stable. 

Oh, but I see you're using the same `skipTests` !!

```
+ mvn clean install -f /mnt/hudson_workspace/workspace/dashboard-next-happy-path/pom.xml -T 4 -U -DskipTests -Dskip-enforce -Dskip-validate-sources
```

Reply:
could we just instead use a predefined image having already those dependencies installed ? (to speed up each run of the jenkins flow)

Reply:
@benoitf: 
> could we just instead use a predefined image having already those dependencies installed ? (to speed up each run of the jenkins flow)

There is a plan to update Jenkins CI node snapshots.

Reply:
yeah but the dependencies still have to come from somewhere, unless we plan to manage our own pulp repo mirrors :D